docs/support/tutorials/python-usage-guide.md

```diff
@@ -0,0 +1,254 @@
+
+	3. An alternative to using Tykky is creating a pip environment
+	   inside a custom Apptainer container. This is a practical choice if, for
+	   example, you know of a suitable ready-made Apptainer or Docker container.
+	   For more information about using Apptainer containers, please see the
+	   related documentation:
+
+		* [Running Apptainer containers](../../computing/containers/run-existing.md)
+		* [Creating Apptainer containers](../../computing/containers/creating.md),
+		including how to convert Docker containers to Apptainer containers.
+
+=== "conda"
+
+	Conda is easy to use and flexible, but it usually creates a huge number of files which
+	is incompatible with shared file systems. The excess of files can cause
+	very slow library imports and,
+	in the worst case, slowdowns in the whole file system. Therefore,
+	**CSC has deprecated the use of conda for direct installations on
+	supercomputers**. However, you can still create and use
+	[containerized](../../computing/containers/overview.md)
+	conda environments.
+
+	1. The most straightforward way to create a containerized conda
+	   environment is by using the [Tykky container
+	   wrapper](../../computing/containers/tykky.md).
+	   To find out how to easily containerize your environment,
+	   see the [Tykky instructions for conda-based
+	   installations](../../computing/containers/tykky.md#conda-based-installations).
+
+	2. An alternative to using Tykky is creating a conda environment
+	   inside a custom Apptainer container. This is a practical choice if, for
+	   example, you know of a suitable ready-made Apptainer or Docker container.
+	   For more information about using Apptainer containers, please see the
+	   related documentation:
+
+		* [Running Apptainer containers](../../computing/containers/run-existing.md)
+		* [Creating Apptainer containers](../../computing/containers/creating.md),
+	  	including how to convert Docker containers to Apptainer containers.
+
+	The [CSC conda tutorial](./conda.md) describes in more detail
+	what conda is and how to use it. Some parts of the tutorial may
+	also be helpful for Tykky installations.
+
+## Python development environments
+
+Python code can be edited with a console-based [text editor directly on the
+supercomputer](./env-guide/text-and-image-processing.md).
+Codes can also be edited on your local machine and copied to the supercomputer
+with [scp](../../data/moving/scp.md) or [graphical file transfer
+tools](../../data/moving/graphical_transfer.md).
+You can also [edit Python scripts in Puhti from your local
+PC](./remote-dev.md) with some code editors like Visual
+Studio Code.
+
+Finally, several graphical programming environments can be used directly on the
+supercomputer, such as Jupyter Notebooks, Spyder and Visual Studio Code, through
+the [Puhti web interface](../../computing/webinterface/index.md).
+
+### Jupyter Notebooks
+
+[Jupyter Notebooks](https://jupyter.org/) allows one to run Python code via a web
```

Comment:
```suggestion
[Jupyter Notebooks](https://jupyter.org/) allows one to run Python code Puhti or Mahti web interface.
```

Reply:
I understood that this was referring to how Jupyter notebooks are opened with a web browser when running them locally. In any case, I removed this part and introduce Jupyter more generally as an "interactive programming environment".